server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -376,6 +376,9 @@ class InteractiveSession(
     }
     uriFuture onFailure { case e => warn("Fail to get rsc uri", e) }
 
+    // Register this class to RSCClient as a session state listener
+    client.get.registerStateListener(this)
```

Comment:
`livy-server` won't get notifications from repl state changes happened before registerStateListener()'s called. `livy-server` will not show the current repl state until repl changes its state once after connection.